package.nix

```diff
@@ -1,27 +1,41 @@
-{ craneLib
+{ pkgs
 , lib
 , stdenv
 , pkg-config
 , openssl
 
-, enableCodeforces ? true
+, enableCodeforces ? false
 , enableOsu ? true
 , ...
 }:
 let
   customizeFeatures = !(enableCodeforces && enableOsu);
-  featureFlags = lib.optionals customizeFeatures (
-    [ "--no-default-features" "--features=core" ]
-    ++ lib.optional enableCodeforces "--features=codeforces"
-    ++ lib.optional enableOsu "--features=osu"
-  );
 in
-craneLib.buildPackage {
+pkgs.rustPlatform.buildRustPackage {
```

Comment:
Add `rustPlatform` into parameter list and use it instead of using `pkgs` directly.